bindings/src/main/java/io/pravega/storage/filesystem/FileSystemStorageFactory.java

```diff
@@ -36,9 +38,24 @@ public FileSystemStorageFactory(FileSystemStorageConfig config, ExecutorService
         this.executor = executor;
     }
 
+    public FileSystemStorageFactory() {
+
+    }
+
     @Override
     public Storage createStorageAdapter() {
         FileSystemStorage s = new FileSystemStorage(this.config);
         return new AsyncStorageWrapper(new RollingStorage(s), this.executor);
```

Comment:
I don't see this executor being shutdown here. Surely it is not up to the AsyncStoreageWrapper right?

Reply:
The executor is passed through all these layers. `ServiceBuilder` is supposed to create and manage it.